src/engine/native/node/native-vm2/src/index.js

```diff
@@ -1,19 +1,193 @@
+    this.childProcesses = new Map();
+
+    this.options = options;
+  }
+
+  executeCommand(command, args, send) {
+    switch (command) {
+      case 'launch-child-process':
+        this.launchSubProcess(...args, send);
+        break;
+      case 'stop-child-process':
+        this.stopSubprocess(...args);
+        break;
+      default:
+        break;
+    }
+  }
+
+  /**
+   * @param {string} processId
+   * @param {string} processInstanceId
+   * @param {string} tokenId
+   * @param {string} scriptString
+   * @param {token} token Token for communicating with universal part
+   * @param { (error: any, response: any) => void} sendToUniversal
+   *
+   */
+  launchSubProcess(
+    processId,
+    processInstanceId,
+    tokenId,
+    scriptString,
+    token,
+    engineScriptExecutionEndpoint,
+    sendToUniversal,
+  ) {
+    // NOTE: maybe just warn
+    if (this.getProcess(processId, processInstanceId))
+      throw new Error('This process is already executing a script');
+
+    let processEntry;
+    try {
+      // TODO: don't hard code port
+      const scriptTask = childProcess.spawn(`node`, [
+        require.resolve('./childProcess.js'),
+        processId,
+        processInstanceId,
+        tokenId,
+        scriptString,
+        token,
+        engineScriptExecutionEndpoint,
+      ]);
+
+      processEntry = {
+        process: scriptTask,
+        sendToUniversal,
+        token,
+        stdout: '',
+        stderr: '',
+      };
+
+      this.setProcess(processId, processInstanceId, processEntry);
+
+      scriptTask.stdout.on('data', (data) => {
+        if (this.options.logChildProcessOutput) console.log(`stdout: ${data}`);
+        processEntry.stdout += data.toString();
+      });
+
+      scriptTask.stderr.on('data', (data) => {
+        if (this.options.logChildProcessOutput) console.error(`stderr: ${data}`);
+        processEntry.stderr += data.toString();
+      });
+
+      scriptTask.on('close', (code) => {
+        processEntry.sendToUniversal(undefined, [
+          {
+            type: 'process-finished',
+            code,
+          },
+        ]);
+      });
+    } catch (e) {
+      // TODO: use proper logger
+      console.error('Failed to start subprocess', e);
+      processEntry.sendToUniversal(undefined, [
+        {
+          type: 'process-finished',
+          code: 1,
+        },
+      ]);
+    }
+  }
+
+  /**
+   * @param {string} processId
+   * @param {string} processInstanceId
+   */
+  stopSubprocess(processId, processInstanceId) {
+    // NOTE: mabybe the onClose function from a process should remove them from the map
+
+    // required by the neo engine
+    if (processId === '*' && processInstanceId === '*') {
+      for (const scriptTask of this.getAllProcesses()) {
+        if (!scriptTask.process.kill())
+          throw new Error(`Failed to kill subprocess ${process.process.pid}`);
+      }
+    }
+
+    const scriptTask = this.getProcess(processId, processInstanceId);
```

Comment:
Shouldn't this be the else path? It seems that this would trigger the early return since a process with id * will most likely not exist but maybe this could be unified in some way to make that clearer

Maybe you could do something like:
```
let toStop : ScriptExecutionType =  [];

if (processId === '*' && processInstanceId === '*') {
  toStop = this.getAllProcesses();
} else {
  const scriptTask = this.getProcess(processId, processInstanceId);
  if (scriptTask) toStop.push(scriptTask);
}

for (const scriptTask of toStop) {
  if (!scriptTask.process.kill()) throw new Error(`Failed to kill subprocess ${process.process.pid}`);
  this.deleteProcess(get the ids somehow);
}
```